package/kedro_viz/models/flowchart.py

```diff
@@ -267,31 +277,30 @@ def add_pipeline(self, pipeline_id: str):
         self.pipelines.add(pipeline_id)
 
     def belongs_to_pipeline(self, pipeline_id: str) -> bool:
-        """Check whether this graph node belongs to a given pipeline_id."""
+        """Checks whether this graph node belongs to a given pipeline_id."""
         return pipeline_id in self.pipelines
 
     def has_metadata(self) -> bool:
-        """Check whether this graph node has metadata.
+        """Checks whether this graph node has metadata.
         Since metadata of a graph node is derived from the underlying Kedro object,
         we just need to check whether the underlying object exists.
         """
         return self.kedro_obj is not None
 
 
-@dataclass
-class GraphNodeMetadata(abc.ABC):
+class GraphNodeMetadata(BaseModel):
     """Represent a graph node's metadata"""
 
 
-@dataclass
 class TaskNode(GraphNode):
     """Represent a graph node of type TASK"""
 
-    modular_pipelines: List[str] = field(init=False)
-    parameters: Dict = field(init=False, default_factory=dict)
+    modular_pipelines: List[str] = []
+    parameters: Dict = {}
     type: str = GraphNodeType.TASK.value
 
-    def __post_init__(self):
+    def __init__(self, **data):
```

Comment:
This will work ok but is not really the right way to do things in pydantic. What I'd recommend is something like this:
```
@validator("namespace")
def set_namespace(cls, values):
    return values["kedro_obj"].namespace

@validator("modular_pipelines")
def set_modular_pipelines(cls, values):
    return self._expand_namespaces(values["kedro_obj"].namespace)
```

[Default factories](https://docs.pydantic.dev/1.10/usage/models/#field-with-dynamic-default-value) are also useful here and work like they do in dataclasses, but can't be used to reference other fields in the same model, hence the need for validators here.